py/selenium/webdriver/safari/service.py

```diff
@@ -42,20 +42,30 @@ def __init__(
         reuse_service=False,
         **kwargs,
     ) -> None:
-        self.service_args = service_args or []
+        if service_args is None:
+            service_args = []
+        self._service_args = service_args
+
         if quiet is not None:
             warnings.warn("quiet is no longer needed to supress output", DeprecationWarning, stacklevel=2)
+        self.reuse_service = reuse_service
 
-        self._reuse_service = reuse_service
         super().__init__(
             executable=executable_path,
             port=port,
             env=env,
             **kwargs,
         )
 
-    def command_line_args(self) -> typing.List[str]:
-        return ["-p", f"{self.port}"] + self.service_args
+    @property
+    def service_args(self):
+        return self._service_args
+
+    @service_args.setter
+    def service_args(self, value):
+        if not isinstance(value, list):
+            raise TypeError("service args must be a list")
+        self._service_args.extend(value)
```

Comment:
It might be just me, but the setter seems more like an append rather than replace/set the value.

Is this the intended behaviour?  Setters for me usually replace the existing value

Reply:
every time when assignment happens to `service_args` I am not replacing the old args, rather I am updating the existing contents of the list. 
e.g. 
```
service = Service(service_args = ["--log", "debug"])
service.service_args = ["--port", 1234]
print(service.service_args)
# prints 
["--log", "debug", "--port", 1234]
```
@titusfortner , should this be the intended behaviour? or should the old list be replaced with new one every time an assignment happens to `service_args`?

Reply:
I don't know the Python convention, but it does feel a little weird to append to something existing with an assignment operator. (They way Ruby does it is so much better 😉). I lean toward replacing instead of adding. @isaulv what do you think?

Reply:
This should definitely not be a `setter` as that creates the expectation that it is setting the value that you pass in. It is not intuitive to users that it will extend. We either need a different method name or we should see about closing this PR.